diff --git a/scrapy/cmdline.py b/scrapy/cmdline.py
index da0e51386..03d3bd773 100644
--- a/scrapy/cmdline.py
+++ b/scrapy/cmdline.py
@@ -96,15 +96,31 @@ def _pop_command_name(argv: List[str]) -> Optional[str]:
     return None
 
 
+branch_coverage_print_header = {
+    "_print_header_1": False,
+    "_print_header_2": False,
+}
+
+
 def _print_header(settings: BaseSettings, inproject: bool) -> None:
     version = scrapy.__version__
     if inproject:
+        branch_coverage_print_header["_print_header_1"] = True
         print(f"Scrapy {version} - active project: {settings['BOT_NAME']}\n")
-
     else:
+        branch_coverage_print_header["_print_header_2"] = True
         print(f"Scrapy {version} - no active project\n")
 
 
+def _print_branch_coverage_cmd():
+    coverage_percentage = (
+        sum(branch_coverage_print_header.values()) / len(branch_coverage_print_header)
+    ) * 100
+    for branch, executed in branch_coverage_print_header.items():
+        print(f"{branch} has been {'executed' if executed else 'missed'}")
+    print(f"Branch coverage: {coverage_percentage}%")
+
+
 def _print_commands(settings: BaseSettings, inproject: bool) -> None:
     _print_header(settings, inproject)
     print("Usage:")
